tests/rptest/services/spark_service.py

```diff
@@ -149,6 +149,28 @@ def escape_identifier(self, table: str) -> str:
     def engine_name():
         return QueryEngineType.SPARK
 
+    def can_run_maintenance(self):
+        return True
+
+    def run_sample_maintenance_task(self, namespace, table) -> None:
+        # Metadata query
+        # https://iceberg.apache.org/docs/1.6.1/spark-queries/#files
+        initial_parquet_files = self.run_query_fetch_one(
+            f"SELECT count(*) FROM {namespace}.{table}.files")[0]
+
+        # Want at least 2 files to be able to assert that optimization did something.
+        assert initial_parquet_files >= 2, f"Expecting at least 2 files, got {initial_parquet_files}"
+
+        # Spark Procedures provided by Iceberg SQL Extensions
+        # https://iceberg.apache.org/docs/1.6.1/spark-procedures/#rewrite_data_files
+        self.run_query_fetch_one(
+            f"CALL `redpanda-iceberg-catalog`.system.rewrite_data_files(\"{namespace}.{table}\")"
+        )
+
+        optimized_parquet_files = self.run_query_fetch_one(
+            f"SELECT count(*) FROM {namespace}.{table}.files")[0]
+        assert optimized_parquet_files < initial_parquet_files, f"Expecting fewer files after optimize, got {optimized_parquet_files}"
```

Comment:
IMO these service classes should only hold operations, not validations. To that end, can we split this into some `count_parquet_files()` and `optimize_parquet_files()`? There are many different kinds of maintenance https://iceberg.apache.org/docs/1.5.1/maintenance/ and it'll be much easier to reuse if we're more descriptive and have smaller, tighter primitives.

Reply:
@andrwng I see what you mean but here I wanted to give more freedom to these maintenance tasks implementation, i.e. it might as well be a metadata only maintenance. I documented the expectations in the code doc https://github.com/redpanda-data/redpanda/blob/9874fa01f80ff843abf586921973932b5b13bc47/tests/rptest/tests/datalake/query_engine_base.py#L70

`optimize_parquet_files` and `count_parquet_files` would work for the 2 engines we have now, do you reckon all of them expose the same primitives so that we could have a single test for all of them?